### PR TITLE
docs: update custom-field React basic example to use binder

### PR DESCRIPTION
**Pull request by web-padawan**
Related to #3212 

Follow-up to #3224 

Replaced the workaround with the proper React form binder and two validators, aligned with the Lit basic example.

### CODE REVIEW

File: frontend/demo/component/custom-field/react/custom-field-basic.tsx [hunk 1]
```diff
@@ -5,37 +5,49 @@ import {
   CustomField,
   type CustomFieldValueChangedEvent,
 } from '@vaadin/react-components/CustomField.js';
-import { differenceInDays, parseISO } from 'date-fns';
+import { differenceInDays, isAfter, parseISO } from 'date-fns';
+import { useForm, useFormPart } from '@vaadin/hilla-react-form';
+import AppointmentModel from 'Frontend/generated/com/vaadin/demo/domain/AppointmentModel';
 
 function Example() {
   const [errorMessage, setErrorMessage] = useState('');
```
Comment (vursen) [inline frontend/demo/component/custom-field/react/custom-field-basic.tsx:13]: This line can be removed – it's not used anymore.

File: frontend/demo/component/custom-field/react/custom-field-basic.tsx [hunk 2]
```diff
@@ -5,37 +5,49 @@ import {
   CustomField,
   type CustomFieldValueChangedEvent,
```
Comment (vursen) [inline frontend/demo/component/custom-field/react/custom-field-basic.tsx:6]: This import can be removed – it's not used anymore.

File: frontend/demo/component/custom-field/react/custom-field-basic.tsx [hunk 3]
```diff
@@ -5,37 +5,49 @@ import {
...
 } from '@vaadin/react-components/CustomField.js';
-import { differenceInDays, parseISO } from 'date-fns';
+import { differenceInDays, isAfter, parseISO } from 'date-fns';
+import { useForm, useFormPart } from '@vaadin/hilla-react-form';
+import AppointmentModel from 'Frontend/generated/com/vaadin/demo/domain/AppointmentModel';
 
 function Example() {
```
Comment (vursen) [inline frontend/demo/component/custom-field/react/custom-field-basic.tsx:12]: The React example is missing this part from the Lit example:

https://github.com/vaadin/docs/blob/52eb5eeb2fd4ddab333fc3b40aa557054eeb0885/frontend/demo/component/custom-field/custom-field-basic.ts#L30-L32

Comment (web-padawan) [reply frontend/demo/component/custom-field/react/custom-field-basic.tsx:9]: Let's discuss this separately. IIRC, using `title` for screen readers is not the best idea.

File: frontend/demo/component/custom-field/react/custom-field-basic.tsx (hunk 1 above)
Comment (web-padawan) [reply frontend/demo/component/custom-field/react/custom-field-basic.tsx]: Thanks, done.

File: frontend/demo/component/custom-field/react/custom-field-basic.tsx (hunk 2 above)
Comment (web-padawan) [reply frontend/demo/component/custom-field/react/custom-field-basic.tsx]: Done.

**Review by vursen — APPROVED** [review]

**Pull request merged**